pkg/secretless/entrypoint/entrypoint.go

```diff
@@ -50,49 +53,77 @@ func StartSecretless(params *SecretlessOptions) {
 		log.Fatalln(err)
 	}
 
-	// Exit signal listener
+	// Optional Performance Profiling
+	handlePerformanceProfiling(params.ProfilingMode, exitListener)
 
-	// Coordinates processes interested in exit signals
-	exitListener := signal.NewExitListener()
+	configChangedChan, err := newConfigChangeChan(
+		params.ConfigFile,
+		params.ConfigManagerSpec,
+		params.FsWatchEnabled,
+	)
+	if err != nil {
+		log.Fatalln(err)
+	}
 
-	// Optional Performance Profiling
+	// Health check: Initialized
+	util.SetAppInitializedFlag()
+	util.SetAppIsLive(false)
 
-	handlePerformanceProfiling(params.ProfilingMode, exitListener)
+	var allServices internal.Service
+
+	configChangedFunc := func(cfg v2.Config) {
+		util.SetAppIsLive(false)
+		// Start Services
+		allServices = proxyservice.NewProxyServices(cfg, availPlugins, logger, evtNotifier)
+		err = allServices.Start()
```

Comment:
Don't we need to stop the old services too?

Reply:
heh - I moved the var outside of the config loop but forgot to actually add the reason why I did that. good catch!